### ISSUE TITLE
False positive use of moved value in math

### ISSUE THREAD

**Issue by KSBilodeau** [intellij-rust/intellij-rust#10468]
## Environment

* **IntelliJ Rust plugin version:**
0.4.194.5382-231

* **Rust toolchain version:**
1.71.0-nightly

* **IDE name and version:**
CLion 2023.1.2

* **Operating system:**
macOS 13.3.1 (a) (22E772610a)

## Problem description
Mathematical operations on copy types incorrectly flagged as a moving operation in various locations.

## Steps to reproduce
All math will produce this if the same variable is used twice in an operation such as:
```rs
pub fn foo(bar: u8) {
    // Second bar is marked as "use of moved value"
    bar + bar;
}
```

I also found that it can trigger without repetition in indexing; however, the behavior was inconsistent and difficult to reproduce.
```rs
pub fn set_starting_peg(&mut self, row: usize, col: usize) -> anyhow::Result<()> {
    if col > row + 1 {
        bail!("({}, {}) is not a valid peg position", &row, &col)
    }

    // Row is marked as "use of moved value"
    self.pegs[row + col] = None;
    self.calc_moves();

    Ok(())
}
```

<img width="695" alt="image" src="https://github.com/intellij-rust/intellij-rust/assets/22333708/2c0fd447-9ad8-4321-9b40-467df04cee02">

<img width="407" alt="Screenshot 2023-05-15 at 12 11 31 PM" src="https://github.com/intellij-rust/intellij-rust/assets/22333708/f5ee6229-68ee-4440-8e75-1fcd8cf6c8f0">

**Comment by vlad20012** [comment]
Duplicate #10427
